app/src/main/java/com/github/swent/echo/viewmodels/ThemeViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.swent.echo.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.swent.echo.ThemePreferenceManager
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+enum class AppTheme {
+    MODE_DAY,
+    MODE_NIGHT
+}
+
+@HiltViewModel
+class ThemeViewModel
+@Inject
+constructor(private val themePreferenceManager: ThemePreferenceManager): ViewModel(){
+    private val _themeUserSetting = MutableStateFlow(AppTheme.MODE_NIGHT)
```

Comment:
wdym by the default value of the system? The default mode of the application is dark mode when the user (re)installs it. It can be changed by the user later on and the changed theme stays that way until the user uninstalls the application. The default value of the variable is set to dark mode and the init block actually fetches the current theme and displays the UI accordingly in the fetched theme of the app.

Reply:
When the user install the app, the default mode of the app is currently dark mode. Instead it should take the value from the phone of the user to avoid inconsistency with the whole system light or dark mode.